graylog2-server/src/main/java/org/graylog2/periodical/DefaultStreamMigrationPeriodical.java

```diff
@@ -0,0 +1,129 @@
+import org.graylog2.streams.StreamService;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Periodical creating the default stream if it doesn't exist.
+ */
+public class DefaultStreamMigrationPeriodical extends Periodical {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultStreamMigrationPeriodical.class);
+
+    private final StreamService streamService;
+    private final StreamRuleService streamRuleService;
+
+    @Inject
+    public DefaultStreamMigrationPeriodical(final StreamService streamService,
+                                            final StreamRuleService streamRuleService) {
+        this.streamService = streamService;
+        this.streamRuleService = streamRuleService;
+    }
+
+    @Override
+    public void doRun() {
+        final ObjectId id = new ObjectId(Stream.DEFAULT_STREAM_ID);
+        final Map<String, Object> fields = ImmutableMap.<String, Object>builder()
+                .put(StreamImpl.FIELD_TITLE, "All messages")
+                .put(StreamImpl.FIELD_DESCRIPTION, "Stream containing all messages")
+                .put(StreamImpl.FIELD_DISABLED, false)
+                .put(StreamImpl.FIELD_CREATED_AT, DateTime.now(DateTimeZone.UTC))
+                .put(StreamImpl.FIELD_CREATOR_USER_ID, "local:admin")
+                .put(StreamImpl.FIELD_MATCHING_TYPE, StreamImpl.MatchingType.DEFAULT.name())
+                .put(StreamImpl.FIELD_DEFAULT_STREAM, true)
+                .build();
+        final Stream stream = new StreamImpl(id, fields, Collections.emptyList(), Collections.emptySet());
+        final StreamRule streamRule = new StreamRuleImpl(
+                ImmutableMap.<String, Object>builder()
+                        .put(StreamRuleImpl.FIELD_TYPE, StreamRuleType.ALWAYS_MATCH.getValue())
+                        .put(StreamRuleImpl.FIELD_FIELD, "timestamp")
+                        .put(StreamRuleImpl.FIELD_INVERTED, false)
+                        .put(StreamRuleImpl.FIELD_STREAM_ID, id)
+                        .put(StreamRuleImpl.FIELD_DESCRIPTION, "Match all messages")
+                        .build());
+        try {
+            streamService.save(stream);
+            streamRuleService.save(streamRule);
+            LOG.info("Successfully created default stream: {}", stream.getTitle());
+        } catch (ValidationException e) {
+            LOG.error("Couldn't create default stream", e);
+        }
+    }
+
+    @Override
+    public boolean runsForever() {
+        return true;
+    }
+
+    @Override
+    public boolean stopOnGracefulShutdown() {
+        return false;
+    }
+
+    @Override
+    public boolean masterOnly() {
+        return true;
+    }
+
+    @Override
+    public boolean startOnThisNode() {
+        try {
+            return streamService.load(Stream.DEFAULT_STREAM_ID) == null;
```

Comment:
I thought about different approaches to this.

The preferred one would be to create a marker in the cluster config, so that the default stream is only created once.

It should still be possible for users to remove the default stream, if they wish so.